Production_scripts/Process_extracts/extract_a&e.R

```diff
@@ -0,0 +1,620 @@
+        "23F",
+        "24",
+        "24A",
+        "24B",
+        "24C",
+        "24D",
+        "25",
+        "25A",
+        "25B",
+        "25C",
+        "25D",
+        "25E",
+        "25F",
+        "25Z",
+        "26",
+        "26A",
+        "26B",
+        "26C",
+        "26D",
+        "26E",
+        "26F",
+        "26G",
+        "26H",
+        "26J",
+        "26K",
+        "26L",
+        "26M",
+        "26Z",
+        "27",
+        "27A",
+        "27B",
+        "27C",
+        "27D",
+        "27E",
+        "27F",
+        "27G",
+        "27H",
+        "27Z",
+        "28",
+        "28A",
+        "28B",
+        "28C",
+        "28D",
+        "28E",
+        "28F",
+        "28Z",
+        "29",
+        "29A",
+        "29B",
+        "29C",
+        "29D",
+        "29E",
+        "29F",
+        "29G",
+        "29H",
+        "29J",
+        "29K",
+        "29L",
+        "29M",
+        "29N",
+        "30",
+        "30A",
+        "30B",
+        "30C",
+        "30D",
+        "30E",
+        "31"
+      )
+    )
+  )
+
+## save outfile ---------------------------------------
+outfile <-
+  ae_clean %>%
+  select(
+    year,
+    recid,
+    record_keydate1,
+    record_keydate2,
+    keyTime1,
+    keyTime2,
+    chi,
+    gender,
+    dob,
+    gpprac,
+    postcode,
+    lca,
+    hscp,
+    location,
+    hbrescode,
+    hbtreatcode,
+    starts_with("diag"),
+    refsource,
+    sigfac,
+    starts_with("ae_"),
+    ends_with("_adm"),
+    cost_total_net,
+    age,
+    ends_with("_cost"),
+    case_ref_number
+  )
+
+
+# ------------------------------------------------------------------------------------------------------------
+
+## CUP Marker ##
+
+# Read in data---------------------------------------
+
+ae_cup_file <- readr::read_csv(
+  file = get_boxi_extract_path(year, "AE_CUP"),
+  col_type = cols(
+    `ED Arrival Date` = col_date(format = "%Y/%m/%d %T"),
+    `ED Arrival Time` = col_time(""),
+    `ED Case Reference Number [C]` = col_character(),
+    `CUP Marker` = col_double(),
+    `CUP Pathway Name` = col_character()
+  )
+) %>%
+  # rename variables
+  rename(
+    record_keydate1 = "ED Arrival Date",
+    keyTime1 = "ED Arrival Time",
+    case_ref_number = "ED Case Reference Number [C]",
+    cup_marker = "CUP Marker",
+    cup_pathway = "CUP Pathway Name"
+  )
+
+
+# Data Cleaning---------------------------------------
+
+ae_cup_clean <- ae_cup_file %>%
+  # Sort for linking and remove any duplicates
+  arrange(record_keydate1, keyTime1, case_ref_number) %>%
+  group_by(record_keydate1, keyTime1, case_ref_number) %>%
+  summarise(
+    cup_marker = first(cup_marker),
+    cup_pathway = first(cup_pathway)
+  ) %>%
+  ungroup()
+
+
+# Join data--------------------------------------------
+
+matched_ae_data <- outfile %>%
+  full_join(ae_cup_clean, by = c("record_keydate1", "keyTime1", "case_ref_number")) %>%
+  arrange(chi, record_keydate1, keyTime1, record_keydate2, keyTime2)
```

Comment:
I think this should just be a `left_join`. The records which have CUP marker should all have a match in the main data but if they don't we don't want to include random bits of data not linked to anything.

```suggestion
matched_ae_data <- outfile %>%
  left_join(ae_cup_clean, by = c("record_keydate1", "keyTime1", "case_ref_number")) %>%
  arrange(chi, record_keydate1, keyTime1, record_keydate2, keyTime2)
```